interval/search.go

```diff
@@ -478,3 +478,66 @@ func maxEnd[V, T any](n *node[V, T], searchEnd T, cmp CmpFunc[T], visit func(*no
 		maxEnd(n.Right, searchEnd, cmp, visit)
 	}
 }
+
+// VisitFunc is called on all values. Returning false will stop iteration, so will an error.
+type VisitFunc[V, T any] func(V, T) (bool, error)
```

Comment:
It seemed redundant.  If you're passing in the `end` value, but can also able to return `false` (now `StopTraversal`) during the visitor, then which is preferred?  IMHO, use of `end` created interface and usability ambiguity because there are now two ways to halt execution of `VisitFunc()`: return `StopTraversal` or compare against `end` and `return nil`.

I'm indifferent to the outcome in the sense that I'm carrying around a local patch for the time being (hence the delay in getting back to you, sorry about that!).  If there was a good reason to have a parallel API or some other code reuse, I could easily be persuaded to change my mind.  Also, this is your repo, so it's ultimately up to you, not me, and I'll make whatever work.  :)

Thank you for the thoughtful review, btw!